test/e2e/k8s/operator.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	if err = (&atlasdeployment.AtlasDeploymentReconciler{
+		Client:           mgr.GetClient(),
+		Log:              logger.Named("controllers").Named("AtlasDeployment").Sugar(),
+		Scheme:           mgr.GetScheme(),
+		AtlasDomain:      config.AtlasDomain,
+		GlobalAPISecret:  config.GlobalAPISecret,
+		ResourceWatcher:  watch.NewResourceWatcher(),
+		GlobalPredicates: globalPredicates,
+		EventRecorder:    mgr.GetEventRecorderFor("AtlasDeployment"),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "AtlasDeployment")
+		return nil, err
+	}
+
+	if err = (&atlasproject.AtlasProjectReconciler{
+		Client:           mgr.GetClient(),
+		Log:              logger.Named("controllers").Named("AtlasProject").Sugar(),
+		Scheme:           mgr.GetScheme(),
+		AtlasDomain:      config.AtlasDomain,
+		ResourceWatcher:  watch.NewResourceWatcher(),
+		GlobalAPISecret:  config.GlobalAPISecret,
+		GlobalPredicates: globalPredicates,
+		EventRecorder:    mgr.GetEventRecorderFor("AtlasProject"),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "AtlasProject")
+		return nil, err
+	}
+
+	if err = (&atlasdatabaseuser.AtlasDatabaseUserReconciler{
+		Client:           mgr.GetClient(),
+		Log:              logger.Named("controllers").Named("AtlasDatabaseUser").Sugar(),
+		Scheme:           mgr.GetScheme(),
+		AtlasDomain:      config.AtlasDomain,
+		ResourceWatcher:  watch.NewResourceWatcher(),
+		GlobalAPISecret:  config.GlobalAPISecret,
+		GlobalPredicates: globalPredicates,
+		EventRecorder:    mgr.GetEventRecorderFor("AtlasDatabaseUser"),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "AtlasDatabaseUser")
+		return nil, err
+	}
+
+	if err = mgr.AddHealthzCheck("health", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		return nil, err
+	}
+	if err = mgr.AddReadyzCheck("check", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		return nil, err
+	}
+	return mgr, nil
+}
+
+type Config struct {
+	AtlasDomain          string
+	EnableLeaderElection bool
+	MetricsAddr          string
+	Namespace            string
+	WatchedNamespaces    map[string]bool
+	ProbeAddr            string
+	GlobalAPISecret      client.ObjectKey
+	LogLevel             string
+	LogEncoder           string
+	LogDir               string
+}
+
+// ParseConfiguration fills the 'OperatorConfig' from the flags passed to the program
+func mergeConfiguration(initCfg *Config) *Config {
+	config := initCfg
+	if config.AtlasDomain == "" {
+		config.AtlasDomain = "https://cloud-qa.mongodb.com/"
+	}
+	if config.MetricsAddr == "" {
+		// random port
+		config.MetricsAddr = ":0"
+	}
+	if config.ProbeAddr == "" {
+		// random port
+		config.ProbeAddr = ":0"
+	}
+	if config.LogLevel == "" {
+		config.LogLevel = "info"
```

Comment:
Maybe we should use "debug" level as default for e2e tests? Up to you though.

Reply:
Nice catch! Done